imod/mf6/utilities/chd_concat.py

```diff
@@ -0,0 +1,48 @@
+from typing import Optional
+
+import xarray as xr
+
+from imod.mf6.chd import ConstantHead
+
+
+def concat_layered_chd_packages(
+    name: str,
+    dict_packages: dict[str, ConstantHead],
+    remove_merged_packages: bool = True,
+) -> Optional[ConstantHead]:
+    """
+
+    Parameters
+    ----------
+    name: str
+        The name of the package that was split over layers.
+        If they are called "chd-1" and so on then set name to "chd
+    dict_packages: dict[str, ConstantHead]
+        dictionary  with package names as key and the packages as values
+    remove_merged_packages: bool = True
```

Comment:
I have some doubts about the ``remove_merged_packages`` option. 

I think you implemented it to avoid double memory usage in the ``GroundwaterFlow.from_imod5_data`` method. However, this happens also in ``concat_layered_chd_packages`` anyway (dataset per layer in list, and concatenated dataset). The duplicate memory use now avoided in ``GroundwaterFlow.from_imod5_data`` is of little use as ``concat_layered_chd_packages`` is called at the end.